bitmark-references/security.markdown

```diff
@@ -0,0 +1,73 @@
+# Security feature of the Bitmark Blockchain
+
+## Transactions
+
+There are two types of transactions:
+* Single signature - for most basic transactions
+* Countersigned - for special transfers
+
+The data in a transaction is packed in a specific order, depending on
+the transaction type.  An Ed25519 signature is performed over all data
+in the packed transaction and appended to form the submission to the
+block chain.
+
+In the case of a countersigned transaction, the signature is performed
+over the packed transaction *including* the first signature to form
+the complete transaction.  While this does make the signatures order
+sensitive, it does mean that the receiver of the transfer can verify
+the sender before signing.
+
+The Ed25519 signature algorithm details can be viewed at: https://ed25519.cr.yp.to/
+
+The important features for the Bitmark Block chain are:
+* high security signatures with collision resistance
+* small 32 byte keys
+* signature is only 64 bytes
+* fast verification
+
+## Transaction Identifiers
+
+The SHA3-256 algorithm is used to derive a 32 byte identifier for each
+transaction.  This secure hash is over all bytes, including signatures
+of the packed transaction.  This is a NIST standard and is the
+replacement for SHA2.  SHA3 is used here because it has protection
+against the length extension attack that can be performed against
+SHA2.
+
+For more details: https://en.wikipedia.org/wiki/SHA-3
+
+The reason for signing over the signature bytes is to protect against
+signature substitution.
+
+The Transaction Identifiers are merged into a single root hash using
+the Merkle Tree Algorithm: https://brilliant.org/wiki/merkle-tree/
+
+Since the tree is sensitive to missing or changed hashes and to the
+order of the hashes and produces a single result it can be used to
+minimise the amount of data that is stored in the block header to
+secure the transactions.  The Bitmark Blockchain used SHA3-256 for the
```

Comment:
Why does `sensitive to the order of hashes` lead to `generate a single result to save disk space`?